source/utilities.cc

```diff
@@ -53,6 +54,222 @@ namespace aspect
+                  //    all --> assign the associated value to all fields
+                  //    background (or bg) --> assign associated value to
+                  //                           the background
+                  // All other keywords are used to assign the correct
+                  // value to the correct position of the property array,
+                  // and should match the list of compositional field names
+                  // found in the Compositional fields subsection.
+                  //
+                  // The user must input a background value.
+                  //
+                  // each entry must have the format:
+                  // <id> : <value>
+
+
+                  // First tease apart the two halves
+                  const std::vector<std::string> split_parts = Utilities::split_string_list (*p, ':');
+
+                  // Ensure that each entry has the correct form.
+                  AssertThrow (split_parts.size() == 2,
+                               ExcMessage ("The format for mapped "
+                                           + field_name
+                                           + "requires that each entry has the "
+                                           "form `<id> : <value>' "
+                                           ", but the entry <"
+                                           + *p
+                                           + "> does not appear to follow this pattern."));
+
+                  // The easy part: get the name of the compositional field
+                  const std::string keyword_labels = split_parts[0];
+
+                  // Ensure that the special keyword "all" was not used when multiple entries exist.
+                  AssertThrow (keyword_labels != "all",
+                               ExcMessage ("There are multiple "
+                                           + field_name
+                                           + " values found, the keyword `all' is not "
+                                           "allowed. Please check your "
+                                           + field_name
+                                           + " list."));
+
+                  // Continue with placing values into the correct positions according to
+                  // the order of id_names found in the Compositional fields subsection list.
+                  if (keyword_labels == "background" || keyword_labels == "bg")
+                    {
+                      // Overwrite the default x_mapped_values to user input.
+                      // Element_0 is reserved for the background.
+                      x_mapped_values[0] = (Utilities::string_to_double(split_parts[1]));
+                    }
+                  else
+                    {
+                      // Ensure that each non-special keyword found is also contained in
+                      // the Compositional fields subsection list, and insert the associated
```

Comment:
I used the isnan functionality because it allows the code to check if an attempt is made to populate a field more than once. I'm not sure that using the find function will do the same thing.